geomesa-features/geomesa-feature-kryo/src/main/scala/org/locationtech/geomesa/features/kryo/json/JsonPathPropertyAccessor.scala

```diff
@@ -97,66 +131,23 @@ object JsonPathPropertyAccessor {
 
   class JsonPropertyAccessorFactory extends PropertyAccessorFactory {
 
-    override def createPropertyAccessor(typ: Class[_],
-                                        xpath: String,
-                                        target: Class[_],
-                                        hints: Hints): PropertyAccessor = {
+    override def createPropertyAccessor(
+        typ: Class[_],
+        xpath: String,
+        target: Class[_],
+        hints: Hints): PropertyAccessor = {
       if (xpath == null || !xpath.startsWith("$.")) {
         null
       } else if (classOf[SimpleFeature].isAssignableFrom(typ)) {
-        JsonPathFeatureAccessor
+        JsonPathPropertyAccessor
       } else if (classOf[SimpleFeatureType].isAssignableFrom(typ)) {
-        JsonPathTypeAccessor
+        JsonPathPropertyAccessor
```

Comment:
Ok, just to double-check, both of these should be JsonPathPropertyAccessor, right?

Reply:
yeah - i just updated this to consolidate the if checks into one statement to make it clearer. As more background, we originally had 2 different accessors, one for simple feature types and one for simple features. the changes to the transforms caused some changes in how these get called in the iterators - specifically that originally the simple feature type one was never called in the iterators but now it does get called for handling json path attributes. that lead to some exceptions in the geotools code that caches and re-tries the last accessor that was used. the fix for that was to make more robust canHandle/get methods for both accessors, at which point they were basically the same class so i consolidated them.